src/to-have-style.ts

```diff
@@ -0,0 +1,78 @@
+import { matcherHint } from 'jest-matcher-utils';
+import { diff } from 'jest-diff';
+import chalk from 'chalk';
+import type { ReactTestInstance } from 'react-test-renderer';
+import type { ImageStyle, StyleProp, TextStyle, ViewStyle } from 'react-native';
+import { StyleSheet } from 'react-native';
+import { checkReactElement, display } from './utils';
+
+type Styles = TextStyle | ViewStyle | ImageStyle;
+type StyleParameter = StyleProp<Styles>;
+
+const getKeys = Object.keys as <T extends object>(obj: T) => Array<keyof T>;
+
+function printoutStyles(styles: Styles) {
+  return getKeys(styles)
+    .sort()
+    .map((prop) => `${prop}: ${display(styles[prop])};`)
+    .join('\n');
+}
+
+/**
+ * Recursively narrows down the properties in received to those with counterparts in expected
+ */
+function narrow(expected: Styles, received: Styles) {
```

Comment:
Pls re-apply adding TS types to the current version of this file.

Reply:
What do you mean?